daffodil-runtime1/src/main/scala/org/apache/daffodil/infoset/InfosetWalker.scala

```diff
@@ -408,14 +413,23 @@ class InfosetWalker private (
     containerIndexStack.setTop(top + 1)
   }
 
+  @inline
+  private def doOutputter(outputterFunc: => Boolean, desc: String, context: ThrowsSDE): Unit = {
+    Try(outputterFunc) match {
+      case Success(true) => // success
+      case Success(false) => context.SDE("Failed to %s", desc)
```

Comment:
Can this case even happen?  Add comment about how.

Reply:
The InfosetInputter API has it so the functions either return a boolean to indicate success/fail, or throw an exception. I'm not sure if that really makes sense, but it is what it is.

We could deprecate that behavior and a change the interface to be void functions and only expect no return value on success and an exception on failure. I'm sure no custom infoset inputters rely returning true/false since we never handled the case before, but it would change the InfosetInputter interface and so its not backwards compatible.

Reply:
Yes, that API is bothersome. They should return Unit (no value) on success, throw on failure. 

They should throw a predictable InfosetInputterException or InfosetOutputterException respectively which encapsulates the real reason exception object.

Reply:
Agreed. I don't know of any custom InfosetOutputters, so probably better to break the API now before any are created. Unless there are objections, I might save that change for another pull request. This was a fairly minor change that was needed to to support failing XML payloads. Improving the API to not use booleans and predicable exception's feels like a larger task.

Reply:
I'm ok with putting this off to a separate change set.

Reply:
Perhaps the version that we expect to deprecate should just Assert.notImplemented(...)